### ISSUE TITLE
Failure to parse manifest in `kubernetes_manifest` resource

### ISSUE THREAD

**Issue by jootten** [hashicorp/terraform-provider-kubernetes#1862]
Hey, there seems to be an issue with lists in `kubernetes_manifest`.

### Terraform Version, Provider Version and Kubernetes Version

```
Terraform version: 1.3.2
Kubernetes provider version: 1.14.0
Kubernetes version: 1.22.12
```

### Affected Resource(s)

- kubernetes_manifest

### Terraform Configuration Files
[resource gist](https://gist.github.com/jootten/b761e17e05dd80251b03f67775dd4ac3)

### Debug Output

[error gist](https://gist.github.com/jootten/9fa5d29a0640c384030c8beb2b1ed643)

### Steps to Reproduce

1. `terraform apply`

### Expected Behavior
Creation of daemonset as specified in the manifest.

### Actual Behavior
Failure to parse the list of volumes under `spec.template.spec.volumes` with `Error: Failed to transform List value into Tuple of different length`.

### Important Factoids

* The resource is created without issues on Kubernetes provider version 2.7.1.
* From version 2.8 to 2.12 a different error occurs: `Error: Failed to morph manifest to OAPI type`.
* The manifest is taken without modifications from Google ([here](https://raw.githubusercontent.com/GoogleCloudPlatform/container-engine-accelerators/master/nvidia-driver-installer/cos/daemonset-preloaded.yaml)).

### References

- A similar issue has been described [here](https://discuss.hashicorp.com/t/terraform-errors-and-how-to-debug-them/44544).

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by jootten** [comment]
Yes, we found that the existing resource could not be updated anymore after upgrading the Kubernetes provider. The solution for us was to destroy the resource with the old Kubernetes provider version and then recreate it after upgrading the Kubernetes version.

**Comment by miguelaferreira** [comment]
I'm also facing this issue trying to apply a DataDog Agent (v2alpha1) custom resource manifest with volume configuration. As soon as the list of volumes has more than 1 element, this issue pops up.

**Comment by miguelaferreira** [comment]
Trying to keep the issue open ...
